tests/ci/run_inject_hash_cpp.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+set -exo pipefail
+
+# Get directory containing this script
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+cd "${DIR}/../.."
+
+# Source common setup
+source tests/ci/common_posix_setup.sh
+
+# Build AWS-LC with C++ inject_hash
+if [[ "$OSTYPE" == "darwin"* ]]; then
+    # macOS only supports shared builds
+    run_build \
+        -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+        -DFIPS=1 \
+        -DFIPS_SHARED=1 \
+        -DBUILD_SHARED_LIBS=1 \
+        -DUSE_CPP_INJECT_HASH=ON \
+        -DCMAKE_OSX_SYSROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk
+else
+    # Linux supports both static and shared builds
+    # if [[ "${BUILD_TYPE}" == "static" ]]; then
+    #     run_build \
+    #         -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+    #         -DFIPS=1 \
+    #         -DFIPS_SHARED=0 \
+    #         -DBUILD_SHARED_LIBS=0 \
+    #         -DUSE_CPP_INJECT_HASH=ON
+    # else
+    #skipping handling of static build for now
+    run_build \
+        -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+        -DFIPS=1 \
+        -DFIPS_SHARED=1 \
+        -DBUILD_SHARED_LIBS=1 \
+        -DUSE_CPP_INJECT_HASH=ON
+    # fi
+fi
+
+cd "${BUILD_ROOT}"
+
+# Test actual FIPS injection
+echo "TESTING INJECT_HASH.CPP WITH EDGE CASES..." 
+./util/fipstools/inject_hash_cpp/inject_hash_cpp || echo "--Expected failure with no args--"
```

Comment:
Should this fail the build if this isn't an error? Also I prefer moving more of this test logic into actual test files like our tools.